ext/node/polyfills/_tls_wrap.ts

```diff
@@ -151,10 +153,20 @@ export class TLSSocket extends net.Socket {
       handle.afterConnect = async (req: any, status: number) => {
         try {
           const conn = await Deno.startTls(handle[kStreamBaseField], options);
+          try {
+            const hs = await conn.handshake();
+            if (hs.alpnProtocol) {
+              tlssock.alpnProtocol = hs.alpnProtocol;
+            } else {
+              tlssock.alpnProtocol = false;
+            }
+          } catch {
+            // Don't interrupt "secure" event if handshake fails
+          }
```

Comment:
Might want to emit an error event here. Can you check what Node.js does?

Reply:
The error event is emitted in the read/write operation.